packages/CodeQuality/src/Rector/For_/ForRepeatedCountToOwnVariableRector.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\CodeQuality\Rector\For_;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Assign;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Stmt\For_;
+use Rector\Rector\AbstractRector;
+use Rector\RectorDefinition\CodeSample;
+use Rector\RectorDefinition\RectorDefinition;
+
+/**
+ * @see \Rector\CodeQuality\Tests\Rector\For_\ForRepeatedCountToOwnVariableRector\ForRepeatedCountToOwnVariableRectorTest
+ */
+final class ForRepeatedCountToOwnVariableRector extends AbstractRector
+{
+    /**
+     * @var string
+     */
+    private const DEFAULT_VARIABLE_COUNT_NAME = 'itemsCount';
+
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Change count() in for function to own variable', [
+            new CodeSample(
+                <<<'PHP'
+class SomeClass
+{
+    public function run($items)
+    {
+        for ($i = 5; $i <= count($items); $i++) {
+            echo $items[$i];
+        }
+    }
+}
+PHP
+,
+                <<<'PHP'
+class SomeClass
+{
+    public function run($items)
+    {
+        $itemsCount = count($items);
+        for ($i = 5; $i <= $itemsCount; $i++) {
+            echo $items[$i];
+        }
+    }
+}
+PHP
+
+            ),
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [For_::class];
+    }
+
+    /**
+     * @param For_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $countInCond = null;
+        $countedValueName = null;
+
+        $this->traverseNodesWithCallable($node->cond, function (Node $node) use (&$countInCond, &$countedValueName) {
+            if (! $node instanceof FuncCall) {
+                return null;
+            }
+
+            if (! $this->isName($node, 'count')) {
+                return null;
+            }
+
+            $countInCond = $node;
+            $valueName = $this->getName($node->args[0]->value);
+
+            if ($valueName === null) {
+                $countedValueName = self::DEFAULT_VARIABLE_COUNT_NAME;
+            } else {
+                $countedValueName = $valueName . 'Count';
+            }
+
```

Comment:
Should the rector make sure that there is no other - already existing variable - in the current scope which might collide with he newly introduced one?

Reply:
Could you provide code example?

How would you name new variable in case of conflict?

Reply:
```php
class SomeClass
 {
     public function run($items)
     {
         $itemsCount = 25;
         for ($i = 5; $i <= count($items); $i++) {
             echo $items[$i];
         }
         If ($itemsCount == 25 ) echo „whoo“;
     }
 }
```

Reply:
I see

Reply:
Covered in https://github.com/rectorphp/rector/pull/2562

Reply:
Perfect, thx